src/hooks/useOpenFiles.tsx

```diff
@@ -0,0 +1,62 @@
+"use client";
+import { createContext, useContext, useEffect, useState } from "react";
+import { LinkProps } from "next/link";
+import { usePathname } from "next/navigation";
+import { explorerFiles, FileType } from "@/components/Explorer";
+
+type OpenFilesContextProps = {
+  openFiles: string[];
+  markFileAsOpen: (tab: string) => void;
+  closeFile: (tabIndex: number) => void;
+  currentOpenFile: () => FileType | null;
+};
+
+const OpenFilesContext = createContext({} as OpenFilesContextProps);
+
+export function OpenFilesProvider({ children }: { children: React.ReactNode }) {
+  const pathName = usePathname();
+
+  const [openFiles, setOpenFiles] = useState<string[]>(() => {
```

Comment:
Instead of `string[]` here, we can use something like `Array<keyof typeof explorerFiles>` so `openFiles` will always include only existent files. This will also remove the necessity to add [this if](https://github.com/diego3g/faladev/pull/10/files#diff-901ab94412318093eb62f153fbc2550b254a8547d5efdf2ace25306a42b53fc1R20-R22) and [this one](https://github.com/diego3g/faladev/pull/10/files#diff-b8da9efcff34359fa61398a983e06f3486a714fb35205e29420027ce63fbae66R22).